### PR TITLE
fix(v10): align off-chain ACK gate with on-chain ST membership + structured rejection reasons

### PR DESCRIPTION
**Pull request by branarakic**
## Summary

Two related fixes uncovered by debugging a stuck rc.11 testnet publish where every core peer's ACK was rejected by the publisher with the same opaque \`Signer X not registered for identity Y\` line. Root cause was an off-chain ACK gate that drifted from the on-chain check in RFC-001, masked by a rejection log that conflated three distinct failure modes into one string.

### 1. \`ec5e490d\` — fix(chain): align verifyACKIdentity with on-chain RFC-001 ACK signer gate

\`KnowledgeAssetsV10._verifyACKSignature\` was rewired by RFC-001 from \`getNodeStakeV10 > 0\` to \`shardingTableStorage.nodeExists(identityId)\`:

\`\`\`solidity
// packages/evm-module/contracts/KnowledgeAssetsV10.sol:777-783
// RFC-001 edge-publish unblocker: ACK signers must be in the active
// sharding table, not merely staked.
require(shardingTableStorage.nodeExists(identityId), \"ACK signer not in sharding table\");
\`\`\`

The off-chain pre-flight in \`evm-adapter.ts:verifyACKIdentity\` exists specifically to skip a doomed on-chain submission, so it must mirror the on-chain gate exactly. It still gated on positive V10 stake, which let sub-\`minimumStake\` operators clear the off-chain ACK quorum and then revert on-chain with \`\"ACK signer not in sharding table\"\` — the publisher saw the off-chain quorum success followed by an opaque contract revert.

Replaces the stake>0 branch with a \`ShardingTableStorage.nodeExists\` read. Drops the V8 archive fallback (the on-chain gate has no V8 fallback either; keeping ours diverged behaviour silently).

Adds three integration tests against the existing Hardhat harness: (a) fully-staked core passes, (b) unrelated wallet not registered as op-key fails, (c) freshly-keyed but unstaked profile fails — the regression case where pre-fix the gate would still pass once any non-zero stake was present.

### 2. \`f44669d5\` — feat(publisher): structured ACK rejection reasons (key/stake/RPC)

The rejection log conflated three different failure modes into the same string:

> \`[ACKCollector] Signer 0xAB... not registered for identity 5 — rejecting ACK from <peer>\`

…which could be: (a) signer is genuinely not an OPERATIONAL_KEY, (b) node is below \`minimumStake\` and not in the sharding table, or (c) the chain RPC threw and we couldn't tell. Each requires a different operator action — same log line. The rc.11 testnet incident burned ~90 min diagnosing a stake-side rejection that looked like a key-side rejection.

Surfaces the three reasons end-to-end:

- \`chain-adapter.ts\`: new \`VerifyACKIdentityReason\` type + \`VerifyACKIdentityResult\` + optional \`verifyACKIdentityDetailed\` method.
- \`evm-adapter.ts\` + \`mock-adapter.ts\`: implement the structured variant; \`verifyACKIdentity\` defers to it for parity.
- \`ack-collector.ts\`: prefer \`verifyIdentityDetailed\` when supplied, surface the reason verbatim, fall back to the legacy boolean callback otherwise.
- \`publisher-runner.ts\` + \`dkg-agent.ts\`: wire the new method through. The agent's wrapper translates a thrown chain-side exception into \`{ valid: false, reason: 'rpc-error' }\` so a filter-expired / rate-limited RPC stops masquerading as a definitive identity rejection (the existing \`try/catch\` returned \`false\` and lost that signal).

Adds three ACKCollector unit tests (structured reason in log, rpc-error distinct from key-not-registered, detailed-takes-precedence-over-boolean). Existing 45 ACK edge-case tests still pass.

## Why ship them together

Patch 2's \`'rpc-error'\` reason exists specifically to make Patch 1's behaviour observable. Without Patch 2 the agent's \`try/catch → return false\` still hides RPC blips behind the (now-correct) ST-membership rejection — same diagnostic confusion as before, just with a slightly different colour.

## Test plan

- [x] \`packages/publisher\` unit suite: \`v10-ack-edge-cases.test.ts\` — 45/45 pass (incl. 3 new structured-verifier cases).
- [x] \`packages/publisher\` unit suite: \`v10-protocol-operations.test.ts\` — 48/48 pass.
- [x] \`packages/chain\` unit + parity: \`evm-adapter.unit.test.ts\` 41/41, \`mock-adapter-parity.test.ts\` 15/15 — parity preserved with new method on both adapters.
- [x] \`packages/agent\` verify-related cases — 2/2 pass.
- [ ] \`packages/chain\` Hardhat integration: 3 new \`evm-adapter.test.ts\` cases — linted clean and follow the existing harness pattern, but require \`spawnHardhatEnv\` to run; CI will exercise them.

## Reproduction context

Triggered while diagnosing why \`Miles\` (rc.11 edge node) couldn't publish to the rc.11 Base Sepolia testnet — every core peer's ACK was rejected with \`Signer X not registered for identity N\`. Probing the chain directly showed all 7 testnet bootstrap identities had \`getNodeStakeV10() = 0\` AND were not in the sharding table — \`getTotalStakeV10()\` was \`0\` network-wide. Seeding 50,001 v9TRAC per identity from a funded test wallet pushed all 7 across \`minimumStake\` and back into the sharding table; the publish then needed BOTH this PR's gate change to make the off-chain pre-flight match the on-chain reality, and the diagnostic split to make future incidents debuggable in minutes instead of hours.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: packages/publisher/src/ack-collector.ts [hunk 1]
```diff
@@ -330,7 +361,24 @@ export class ACKCollector {
...
-          if (this.deps.verifyIdentity) {
+          // Prefer the detailed verifier — surfaces the specific failing
+          // gate in the rejection log so operators can tell apart "this
+          // signer is genuinely not registered" (operator-side) from
+          // "the node is registered but has not crossed minimumStake"
+          // (operator-side, different action) from "we couldn't reach
+          // the chain to check" (infra-side, retryable). Pre-PR every
+          // failure surfaced as the same "not registered" string.
+          if (this.deps.verifyIdentityDetailed) {
+            const verdict = await this.deps.verifyIdentityDetailed(recoveredAddress, identityId);
```
[the remark this answers is not in the document]
Comment (branarakic) [reply packages/publisher/src/ack-collector.ts:372]: Resolved as part of the rc.12 review-consolidation sweep on #716.

This thread's resolution is recorded in the [audit comment](https://github.com/OriginTrail/dkg/pull/716#issuecomment-4553720673) on #716 — see the **#711** section for the addressing commit set and any per-thread notes.

Source PR will auto-close on #716 merge; resolving now to keep the trail clean. If the concern is still live on current `release/rc.12` HEAD code (not on the original per-PR diff Codex saw), please reopen against #716 directly.

File: packages/cli/src/publisher-runner.ts [hunk 2]
```diff
@@ -329,6 +333,12 @@ function createV10ACKProviderForPublisher(
     sendP2P: transport.sendP2P,
     getConnectedCorePeers: transport.getConnectedCorePeers,
     verifyIdentity: async (recoveredAddress: string, claimedIdentityId: bigint) => chain.verifyACKIdentity!(recoveredAddress, claimedIdentityId),
+    // Prefer the structured verifier when the chain adapter exposes it
+    // so the rejection log can report the specific failing gate.
+    ...(typeof chain.verifyACKIdentityDetailed === 'function' ? {
+      verifyIdentityDetailed: async (recoveredAddress: string, claimedIdentityId: bigint) =>
```
[the remark this answers is not in the document]
Comment (branarakic) [reply packages/cli/src/publisher-runner.ts:339]: Resolved as part of the rc.12 review-consolidation sweep on #716.

This thread's resolution is recorded in the [audit comment](https://github.com/OriginTrail/dkg/pull/716#issuecomment-4553720673) on #716 — see the **#711** section for the addressing commit set and any per-thread notes.

Source PR will auto-close on #716 merge; resolving now to keep the trail clean. If the concern is still live on current `release/rc.12` HEAD code (not on the original per-PR diff Codex saw), please reopen against #716 directly.

File: packages/chain/src/mock-adapter.ts [hunk 3]
```diff
@@ -812,14 +813,27 @@ export class MockChainAdapter implements ChainAdapter {
...
+   */
+  async verifyACKIdentityDetailed(
+    recoveredAddress: string,
+    claimedIdentityId: bigint,
+  ): Promise<VerifyACKIdentityResult> {
     const normalizedAddress = recoveredAddress.toLowerCase();
     for (const [addr, id] of this.identities) {
       if (id === claimedIdentityId && addr.toLowerCase() === normalizedAddress) {
-        return true;
+        return { valid: true };
```
[the remark this answers is not in the document]
Comment (branarakic) [reply packages/chain/src/mock-adapter.ts:833]: Resolved as part of the rc.12 review-consolidation sweep on #716.

This thread's resolution is recorded in the [audit comment](https://github.com/OriginTrail/dkg/pull/716#issuecomment-4553720673) on #716 — see the **#711** section for the addressing commit set and any per-thread notes.

Source PR will auto-close on #716 merge; resolving now to keep the trail clean. If the concern is still live on current `release/rc.12` HEAD code (not on the original per-PR diff Codex saw), please reopen against #716 directly.

**Comment by branarakic** [comment]
Already merged into `release/rc.12` in 25223b14; reaches `main` via #716. Closing as superseded — the open state against `main` is a side effect of merging into a non-base branch.
